source/core/module_coreHelpers.js

```diff
@@ -56,15 +56,18 @@
             fpSync = fn;
         };
 
-        Module.coreHelpers.utf8ArrayToStringWithNull = function (u8Array, idxIn) {
+        // Takes Vireo Strings that are UTF-8 encoded strings with known length and returns a JS string
+        Module.coreHelpers.sizedUtf8ArrayToJSString = function (u8Array, startIndex, length) {
             /* eslint-disable no-continue, no-plusplus, no-bitwise */
             var u0, u1, u2, u3, u4, u5;
-            var idx = idxIn;
+            var idx = startIndex;
+            var lastVisitedIndex = startIndex + length;
```

Comment:
Fixed the first case.

The second case of handling invalid UTF-8 code points I opened a separate issue that I'll add more detail to https://github.com/ni/VireoSDK/issues/283

One thing I wanted to point out is that as-is the algorithm shouldn't result in an error. Out of bounds access in JS results in the value of `undefined` and the code [as used from emscripten](https://github.com/rajsite/emscripten-sdk/blob/ee4476d49228259c27847d7a7794906368fa99db/emscripten/master/src/preamble.js#L549) would perform an `undefined & 63` which causes JS to coerce `undefined` to `0`.

And with strings in the middle of an array it may result in corrupt output reading a couple of extra bytes out of the array.

Not saying this is a good approach to handling invalid UTF-8 code points, but as is it shouldn't cause fatal exceptions until a consistent way for handling that case is determined.